server/data_model/src/lib.rs

```diff
@@ -98,6 +98,7 @@ pub struct ImageInformation {
     pub run_strs: Vec<String>,
     pub image_hash: String,
     pub version: ImageVersion, // this gets updated when the hash changes
+    pub image_uri: String,
```

Comment:
Since the image_uri is optional for open-source, I think we should make it optional so that the code has to handle it as optional. This will also make it so that deserializing from the state store an image infomation that does not have the image_uri will not fail.

```suggestion
    pub image_uri: Option<String>,
```

Reply:
serde default might still be needed, I am unsure.

Reply:
Do we need this image_uri at all? If you have the image hash, compute graph, namespace, and image name - we should be able to fetch the uri from a service right? 

For OSS - it's upto the operator to start them, and assuming we have code in the executor to report the image hash and name, the task creator would be able to assign functions to the container.

In case we are not building a service which will have the coordinates of the URI it makes sense to encode this in the public API.